crates/bevy_ui/src/widget/text.rs

```diff
@@ -3,19 +3,29 @@ use bevy_asset::Assets;
 use bevy_ecs::{
     entity::Entity,
     query::{Changed, Or, With},
-    system::{Local, ParamSet, Query, Res, ResMut},
+    system::{Commands, Local, ParamSet, Query, Res, ResMut},
 };
+use bevy_ecs::{prelude::Component, reflect::ReflectComponent};
 use bevy_math::Vec2;
+use bevy_reflect::Reflect;
 use bevy_render::texture::Image;
 use bevy_sprite::TextureAtlas;
-use bevy_text::{DefaultTextPipeline, Font, FontAtlasSet, Text, TextError};
+use bevy_text::{DefaultTextPipeline, Font, FontAtlasSet, Text, TextError, TextSection};
 use bevy_window::{WindowId, Windows};
+use unicode_bidi::BidiInfo;
 
 #[derive(Debug, Default)]
 pub struct QueuedText {
     entities: Vec<Entity>,
 }
 
+/// Corrected text data after applying the Unicode Bidirectional Algorithm
+#[derive(Component, Debug, Default, Clone, Reflect)]
+#[reflect(Component)]
+pub(crate) struct BidiCorrectedText {
+    pub sections: Vec<TextSection>,
+}
```

Comment:
Why is this defined twice?

Reply:
This struct is crate internal - the user shouldnt even know it exists if using the bevy "standard text implementation" to reduce confusion. Same for the systems (text_system and extract_text2d_sprite) (which also update the BidiCorrectedText components). Sadly we have 2 crates atm - one for Text and one for UI Text therefore the doublication.

Reply:
Ah ok, that's... not optimal, but I guess the only way to do it then without bigger refactors